src/backend/gl/src/queue.rs

```diff
@@ -187,8 +187,10 @@ impl CommandQueue {
         let extent = swapchain.extent;
 
         // Wait for rendering to finish
-        unsafe {
-            gl.wait_sync(self.presentation_fence.unwrap(), 0, glow::TIMEOUT_IGNORED);
+        if self.share.private_caps.sync {
```

Comment:
can we refactor this a bit in the following way: if `private_caps.sync` is `false`, we should not create the `presentation_fence`, hence leaving it `None`. Then here we'd just do `if let Some(fence) = self.presentation_fence { ... }`